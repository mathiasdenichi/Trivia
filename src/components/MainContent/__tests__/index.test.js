import React from 'react'
import { shallow, configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import MainHeader from '../'

configure({ adapter: new Adapter() })

describe('Render trivia quiz questions', () => {
  const wrapper = mount(<MainHeader />)
  const skin = shallow(<MainHeader />)

  it('Renders and matches a snap shot.', () => {
    expect(skin).toMatchSnapshot()
  })

  it('Checks an answer to the user inputed bool, and sets state and page number', () => {
    wrapper.instance().checkAnswer()
    wrapper.setState({
      questionPage: 1,
    })
    expect(wrapper.state().questionBank).toBeTruthy()
    expect(wrapper.state().questionPage).toBeGreaterThan(0)
  })
})
