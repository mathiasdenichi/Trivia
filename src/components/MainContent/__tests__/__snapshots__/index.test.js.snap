// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render trivia quiz questions Renders and matches a snap shot. 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainHeader
    fetchQuestions={[Function]}
    questions={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="intro"
      >
        <h1>
          Welcome to the Trvia Challenge
        </h1>
        <h2>
          Can you score 100%?
        </h2>
        <div
          id="questionBox"
        >
          <p>
            You will be presented with 10 True or False Questions
          </p>
        </div>
        <div
          id="nav"
        >
          <button
            id="begin"
            onClick={[Function]}
          >
            Begin
          </button>
        </div>
      </div>,
      "id": "appContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Welcome to the Trvia Challenge
          </h1>,
          <h2>
            Can you score 100%?
          </h2>,
          null,
          <div
            id="questionBox"
          >
            <p>
              You will be presented with 10 True or False Questions
            </p>
          </div>,
          <div
            id="nav"
          >
            <button
              id="begin"
              onClick={[Function]}
            >
              Begin
            </button>
          </div>,
        ],
        "className": "intro",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to the Trvia Challenge",
          },
          "ref": null,
          "rendered": "Welcome to the Trvia Challenge",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Can you score 100%?",
          },
          "ref": null,
          "rendered": "Can you score 100%?",
          "type": "h2",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              You will be presented with 10 True or False Questions
            </p>,
            "id": "questionBox",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "You will be presented with 10 True or False Questions",
            },
            "ref": null,
            "rendered": "You will be presented with 10 True or False Questions",
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              id="begin"
              onClick={[Function]}
            >
              Begin
            </button>,
            "id": "nav",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Begin",
              "id": "begin",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Begin",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="intro"
        >
          <h1>
            Welcome to the Trvia Challenge
          </h1>
          <h2>
            Can you score 100%?
          </h2>
          <div
            id="questionBox"
          >
            <p>
              You will be presented with 10 True or False Questions
            </p>
          </div>
          <div
            id="nav"
          >
            <button
              id="begin"
              onClick={[Function]}
            >
              Begin
            </button>
          </div>
        </div>,
        "id": "appContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Welcome to the Trvia Challenge
            </h1>,
            <h2>
              Can you score 100%?
            </h2>,
            null,
            <div
              id="questionBox"
            >
              <p>
                You will be presented with 10 True or False Questions
              </p>
            </div>,
            <div
              id="nav"
            >
              <button
                id="begin"
                onClick={[Function]}
              >
                Begin
              </button>
            </div>,
          ],
          "className": "intro",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to the Trvia Challenge",
            },
            "ref": null,
            "rendered": "Welcome to the Trvia Challenge",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Can you score 100%?",
            },
            "ref": null,
            "rendered": "Can you score 100%?",
            "type": "h2",
          },
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                You will be presented with 10 True or False Questions
              </p>,
              "id": "questionBox",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "You will be presented with 10 True or False Questions",
              },
              "ref": null,
              "rendered": "You will be presented with 10 True or False Questions",
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                id="begin"
                onClick={[Function]}
              >
                Begin
              </button>,
              "id": "nav",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Begin",
                "id": "begin",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Begin",
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
